src/groovy/heim/session/SessionCollector.groovy

```diff
@@ -0,0 +1,94 @@
+package heim.session
+
+import groovy.util.logging.Log4j
+import heim.tasks.JobTasksService
+import heim.tasks.TaskState
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+/**
+ * Created by piotrzakrzewski on 22/10/15.
+ */
+@Log4j
+@Component
+class SessionCollector {
+
+    def SessionCollector(){
+        new Timer().schedule({
+            garbageCollection()
+        } as TimerTask, collectionInterval, collectionInterval)
+    }
+
+    @Autowired
+    private SessionService sessionService
+
+    @Autowired
+    private JobTasksService jobTasksService
+
+
+    private final Integer collectionInterval = 3 * 60 * 1000 // 3 min
+    private final Integer sessionLifeSpan = 10 * 60 * 1000 // 10 min
+
+    private HashMap<UUID,Date> sessionLastActivity = new HashMap<UUID,Date>()
+    private HashMap<UUID,List> sessionTasks = new HashMap<UUID,List>()
```

Comment:
You're duplicating here state that exists elsewhere (namely in `JobTasksService`).